ibm/service/satellite/data_source_ibm_satellite_storage_configuration.go

```diff
@@ -0,0 +1,122 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package satellite
+
+import (
+	"fmt"
+
+	"github.com/IBM-Cloud/container-services-go-sdk/kubernetesserviceapiv1"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceIBMSatelliteStorageConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMSatelliteStorageConfigurationRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The Location Name.",
+			},
+			"config_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the Storage Configuration.",
+			},
+			"config_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Version of the Storage Configuration.",
+			},
+			"storage_template_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The Storage Template Name.",
+			},
+			"storage_template_version": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The Storage Template Version.",
+			},
+			"user_config_parameters": {
+				Type:        schema.TypeMap,
+				Computed:    true,
+				Description: "The storage configuration parameters depending on the storage template.",
+			},
+			"user_secret_parameters": {
```

Comment:
Same as below

Reply:
This because we're unable to get the secrets from the server side, I take it as an input through terraform and store it in the state file

Reply:
Generally users may not send them in tf files for datasources. You can completed remove this from schema

Reply:
okay done